Index: Chapter04_For.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Chapter04_For.java b/Chapter04_For.java
new file mode 100644
--- /dev/null	(date 1661400982321)
+++ b/Chapter04_For.java	(date 1661400982321)
@@ -0,0 +1,52 @@
+public class Chapter04_for {
+    public static void main(String[] args) {
+
+
+
+        // for문의 구조
+/*
+            for (초기치; 조건문; 증가치) {
+                ...
+            }
+*/
+
+
+        // 예제 (1)
+            String[] numbers = { "one", "two", "three" };
+                for (int i=0; i<numbers.length; i++) {
+                    System.out.println(numbers[i]);
+                }
+
+
+            int[] marks = { 90, 25, 67, 45, 80 };
+                for(int i=0; i<marks.length; i++) {
+                    if(marks[i] >= 60) {
+                        System.out.println((i+1) + "번 학생은 합격입니다.");
+                    } else {
+                        System.out.println((i+1) + "번 학생은 불합격입니다.");
+                    }
+                }
+
+
+        // for와 continue
+            int[] marks2 = { 90, 25, 67, 45, 80 };
+                for(int i=0; i< marks2.length; i++) {
+                    if(marks2[i] < 60) {    // 60점 미만인 학생인 경우 출력 부분 수행하지 않고 for문으로 돌아감
+                        continue;
+                    }
+                    System.out.println((i+1) + "번 학생 축하합니다. 합격입니다.");
+                }
+
+
+        // 이중 for문
+            for(int i=2; i<10; i++) {   // 2부터 9까지 반복 수행
+                for(int j=1; j<10; j++) {   // 1부터 9까지 반복 수행
+                    System.out.print(i*j+" ");
+                }
+                System.out.println("");
+            }
+
+
+
+    }
+}
\ No newline at end of file
Index: Chapter04_While.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Chapter04_While.java b/Chapter04_While.java
new file mode 100644
--- /dev/null	(date 1661400982323)
+++ b/Chapter04_While.java	(date 1661400982323)
@@ -0,0 +1,66 @@
+public class Chapter04_while {
+    public static void main(String[] args) {
+
+
+
+        // while (조건문이 참인 동안 while문의 반복 수행)
+/*
+            while (조건문) {
+            <수행할 문장1>;
+            <수행할 문장2>;
+            <수행할 문장3>;
+            ...
+            }
+*/
+
+
+        // Q. 열 번 찍어 안 넘어 가는 나무 없다
+            int treeHit = 0;
+            while (treeHit < 10) {
+                treeHit++;
+                System.out.println("나무를 " + treeHit + "번 찍었습니다.");
+                if (treeHit == 10) {
+                    System.out.println("나무가 넘어갑니다.");
+                }
+            }
+
+
+        // 무한 루프 (Loop)
+/*
+            while (true) {
+                System.out.println("Ctrl-C를 눌러야 while문을 빠져 나갈 수 있습니다.");
+            }
+*/
+
+
+        // while문 빠져나가기 (break)
+            int coffee = 10;
+            int money = 300;
+
+            while (money > 0) {
+                System.out.println("돈을 지불했으니 커피를 줍니다.");
+                coffee--;
+                System.out.println("남은 커피의 양은 " + coffee + "입니다.");
+
+                    if (coffee == 0) { // coffee가 0이 되면 if문 실행 -> break 호출 -> while문 빠져나감
+                        System.out.println("커피가 다 떨어졌습니다. 판매를 중지합니다");
+                        break;
+                    }
+            }
+
+
+        // while문 조건문으로 돌아가기 (continue)
+            int a = 0;
+
+            while (a < 10) {
+                a++;
+                if (a % 2 == 0) {   // 짝수인 경우 continue 수행 -> while문으로 돌아감
+                    continue;
+                }
+                System.out.println(a);
+            }
+
+
+
+    }
+}
Index: Chapter04_Switch_Case.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Chapter04_Switch_Case.java b/Chapter04_Switch_Case.java
new file mode 100644
--- /dev/null	(date 1661400982322)
+++ b/Chapter04_Switch_Case.java	(date 1661400982322)
@@ -0,0 +1,61 @@
+public class Chapter04_switch_Case {
+    public static void main(String[] args) {
+
+
+
+        // switch/case문의 구조
+/*
+            switch(입력변수) {
+                case 입력값1: ...
+                    break;
+                case 입력값2: ...
+                    break;
+                ...
+                default: ...
+                    break;
+            }
+*/
+
+
+        // 예제 (1)
+            int month = 15;
+            String monthString = "";
+                switch (month) {
+                    case 1:  monthString = "January";
+                        break;
+                    case 2:  monthString = "February";
+                        break;
+                    case 3:  monthString = "March";
+                        break;
+                    case 4:  monthString = "April";
+                        break;
+                    case 5:  monthString = "May";
+                        break;
+                    case 6:  monthString = "June";
+                        break;
+                    case 7:  monthString = "July";
+                        break;
+                    case 8:  monthString = "August";
+                        break;
+                    case 9:  monthString = "September";
+                        break;
+                    case 10: monthString = "October";
+                        break;
+                    case 11: monthString = "November";
+                        break;
+                    case 12: monthString = "December";
+                        break;
+                    default: monthString = "Invalid month"; // 숫자가 아닌 다른 값이 저장되면 default 문장 출력
+                        break;
+                }
+
+            System.out.println(monthString);
+
+
+        // switch/case문은 if else문으로 변경 가능, if else문은 switch/case문으로 변경 불가능
+        // 입력 값이 정형화 되있는 경우 switch/case문 가독성 좋음
+
+
+
+    }
+}
Index: Chapter04_If.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Chapter04_If.java b/Chapter04_If.java
new file mode 100644
--- /dev/null	(date 1661400982322)
+++ b/Chapter04_If.java	(date 1661400982322)
@@ -0,0 +1,90 @@
+import java.util.ArrayList;
+
+public class Chapter04_if {
+    public static void main(String[] args) {
+
+
+
+        // if (조건문) 참이면 조건문 수행, 거짓이면 else 문에 속한 문장 수행
+/*
+            if (조건문) {
+                <수행할 문장1>;
+                <수행할 문장2>;
+            } else {
+                <수행할 문장A>;
+                <수행할 문장B>;
+            }
+*/
+
+
+        // 연산자
+            int x = 3; int y = 2;
+            System.out.println(x > y); // true
+            System.out.println(y < x); // false
+            System.out.println(x == y); // false
+            System.out.println(x != y); // true
+
+
+        // Q1. 3000원의 이상의 돈을 가지고 있으면 택시, 아니면 걸어가라
+            int money = 2000;
+
+            if (money >= 3000) {
+                System.out.println("택시타기");
+            } else {
+                System.out.println("걸어가기");    // 조건문이 거짓이므로 else문 수행
+            }
+
+
+        // x || y (or), x와 y 둘 중 하나가 true면 true
+        // x && y (and), x와 y 모두가 true면 true
+        // !x (not), x가 거짓이면 true
+
+
+        // contains (해당 아이템 조사)
+            boolean card = true;
+            ArrayList<String> pocket = new ArrayList<>();
+            pocket.add("paper");
+            pocket.add("handphone");
+
+            if (pocket.contains("handphone")) { // pocket 안에 handphone이 있으므로 조건문 수행
+                System.out.println("택시타기");
+            } else {
+                System.out.println("걸어가기");
+            }
+
+
+        // else if (다중 조건 판단)
+/*
+            if (조건문) {
+                <수행할 문장1>
+                <수행할 문장2>
+                ...
+            } else if (조건문) {
+                <수행할 문장1>
+                <수행할 문장2>
+                ...
+            } else if (조건문) {
+                <수행할 문장1>
+                <수행할 문장2>
+                ...
+                ...
+            } else {
+               <수행할 문장1>
+               <수행할 문장2>
+               ...
+           }
+*/
+
+
+            if (pocket.contains("money")) { // 조건문이 거짓이므로 2번째 else if문으로 넘어감
+                System.out.println("택시타기");
+            } else if (card) {              // card = true 이므로 2번째 else if문 수행
+                System.out.println("버스타기");
+            } else {
+                System.out.println("걸어가기");
+            }
+
+
+
+    }
+}
\ No newline at end of file
Index: Chapter04_ForEach.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Chapter04_ForEach.java b/Chapter04_ForEach.java
new file mode 100644
--- /dev/null	(date 1661400982321)
+++ b/Chapter04_ForEach.java	(date 1661400982321)
@@ -0,0 +1,34 @@
+import java.sql.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Chapter04_forEach {
+    public static void main(String[] args) {
+
+
+
+        // for each문의 구조 (루프를 돌릴 수 있는 자료형 즉, 배열 및 ArrayList만 가능)
+/*
+        for (type var: iterate) {
+            body-of-loop
+        }
+*/
+
+
+        // for each
+        String[] numbers = { "one", "two", "three" };
+        for(String number:numbers) {
+            System.out.println(number);
+        }
+
+
+        // 예제 (1)
+        ArrayList<String> num = new ArrayList<>(Arrays.asList("one", "two", "three"));
+        for(String number:num) {
+            System.out.println(num);
+        }
+
+
+
+    }
+}
